bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/discovery/BondDiscoveryService.java

```diff
@@ -94,7 +96,7 @@ protected synchronized void startScan() {
                 for (final String deviceId : deviceList) {
                     BondDevice thisDevice = api.getDevice(deviceId);
                     String deviceName;
-                    if (thisDevice != null && (deviceName = thisDevice.name) != null) {
+                    if ((deviceName = thisDevice.name) != null) {
```

Comment:
`thisDevice` can't be null there? 
That means api.getDevice(deviceId) can't return null?

Reply:
Yes. It returns a BondDevice or it throws `BondException`
```
    public BondDevice getDevice(String deviceId) throws BondException {
        String json = request("/v2/devices/" + deviceId);
        logger.trace("BondHome device info : {}", json);
        try {
            return Objects.requireNonNull(gson.fromJson(json, BondDevice.class));
        } catch (JsonParseException e) {
            logger.debug("Could not parse device {}'s JSON '{}'", deviceId, json, e);
            throw new BondException("@text/offline.comm-error.unparseable-response");
        }
    }
```